bids/layout/config/bids.json

```diff
@@ -80,10 +96,11 @@
     ],
 
     "default_path_patterns": [
-        "sub-{subject}[/ses-{session}]/{datatype<anat>|anat}/sub-{subject}[_ses-{session}][_acq-{acquisition}][_ce-{ceagent}][_rec-{reconstruction}]_{suffix<T1w|T2w|T1rho|T1map|T2map|T2star|FLAIR|FLASH|PDmap|PD|PDT2|inplaneT[12]|angio>}{extension<.nii|.nii.gz|.json>|.nii.gz}",
+        "sub-{subject}[/ses-{session}]/{datatype<anat>|anat}/sub-{subject}[_ses-{session}][_acq-{acquisition}][_ce-{ceagent}][_rec-{reconstruction}][_part-{part}]_{suffix<T1w|T2w|T1rho|T1map|T2map|T2star|FLAIR|FLASH|PDmap|PD|PDT2|inplaneT[12]|angio>}{extension<.nii|.nii.gz|.json>|.nii.gz}",
```

Comment:
When `part` is missing, this means that the data are magnitude. I am not sure if it's possible to have a default value, but I did notice the pattern `{extension<.nii|.nii.gz|.json>|.nii.gz}` in this line. Is the final `|.nii.gz` outside of the `<>` statement referring to a default?

Reply:
Yes. Interpret that as extension can be `.nii`, `.nii.gz` or `.json`, and if absent use `.json`.

I don't think we should necessarily add `part-mag` by default. A file needs to have an extension, it does not need a `part` entity.

Reply:
Ah, I was hoping it was more like, if you search for part=mag, it would return files where part isn't specified.

Reply:
Oh... This is for generating filenames. For that query you should be able to say `layout.get(part=['mag', None])` (or something like that).

Reply:
Thanks! I think I'll need to feed that back into associated pipelines instead of here then (e.g., https://github.com/nipreps/niworkflows/issues/591).